src/utils/gif_utils.js

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export function generateUUID() {
+    let time = new Date().getTime();
+    const id = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (x) => {
+        const d = (time + (Math.random() * 16)) % 16 | 0;
+        time = Math.floor(time / 16);
+        return (x === 'x' ? d : ((d & 3) | 8)).toString(16);
+    });
+    return id;
+}
+
+export function setCookie({name, value, exseconds}) {
+    if (typeof document === 'undefined') {
+        return;
+    }
+    let expires = '';
+    if (exseconds) {
+        const d = new Date();
+        d.setTime(d.getTime() + (exseconds * 1000));
+        expires = `expires=${d.toUTCString()};`;
+    }
+
+    document.cookie = `${name}=${value};${expires}domain=.gfycat.com;path=/;`;
```

Comment:
Independent of the security concerns above, I don't think domain=.gfycat.com even works, since this application isn't being served from .gfycat.com.

Reply:
I've removed the cookie utilities.